src/rgw/rgw_lua_request.cc

```diff
@@ -808,9 +724,11 @@ int execute(
   lua_pushlightuserdata(L, const_cast<char*>(op_name));
   lua_pushcclosure(L, RequestLog, FIVE_UPVALS);
   lua_rawset(L, -3);
+  
+  //TODO: Map is still readonly from lua.
+  using Type = std::map<std::string, std::string>;
+  create_metatable<StringMapMetaTable<Type, StringMapWriteableNewIndex<Type>>>(L, true, &(test->rgw_map_s));
```

Comment:
where is "test" coming from?

Reply:
Passed to 'execute' function from rgw_process.cc.

[1] https://github.com/ceph/ceph/pull/42779/files#diff-7acbbcf700f678720d7d5d9330a084be3b3c310083982997a9dc2dbe3ad4e7dcR345

Reply:
ok. so "test" is not a great name :-)
what's wrong with "background"?

Reply:
The map passed here is not accessible from the Lua script (says its read only). still trying to figure it out hence the temporary name, I'll rename it. :smile:

It seems like 'create_metatable' is behaving incorrectly when called for the second time with the same Lua state.

Reply:
each request has its own lua state. where is it called twice for the sam estate?

Reply:
the function 'create_metatable' is used first time to create the 'Request' table (line 714 [1]).
then Im using it again (with the same state) to create 'RGW' table ( the background cpp map) ,


[1] https://github.com/ceph/ceph/pull/42779/files/8c6415c34653cb31802e8d0f6b2e7930820fe5b7..38fba539f7db19c78985fedaf47d65c99cb8b231#diff-3ab589681dc4ac6c175178a63c853b7586d379bfcba783f2ccc4dd015ae64ac5R714